bc_obps/compliance/service/penalty_calculation_service.py

```diff
@@ -329,4 +452,153 @@ def create_penalty(cls, obligation: ComplianceObligation) -> CompliancePenalty:
+            monthly_compounding = (accumulated_penalty + accumulated_compounding) * monthly_rate
+            accumulated_penalty += penalty_amount
+            accumulated_compounding += monthly_compounding
+            total_penalty = accumulated_penalty + accumulated_compounding
+
+            if persist_penalty_data:
+                CompliancePenaltyAccrual.objects.create(
+                    compliance_penalty=compliance_penalty_record,
+                    date=accrual_date,
+                    interest_rate=monthly_rate.quantize(Decimal('0.0001'), rounding=ROUND_HALF_UP),
+                    daily_penalty=penalty_amount.quantize(Decimal('0.01'), rounding=ROUND_HALF_UP),
+                    daily_compounded=monthly_compounding.quantize(Decimal('0.01'), rounding=ROUND_HALF_UP),
+                    accumulated_penalty=accumulated_penalty.quantize(Decimal('0.01'), rounding=ROUND_HALF_UP),
+                    accumulated_compounded=accumulated_compounding.quantize(Decimal('0.01'), rounding=ROUND_HALF_UP),
+                )
+
+        if persist_penalty_data:
+            compliance_penalty_record.penalty_amount = total_penalty.quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
+            compliance_penalty_record.save()
+            return compliance_penalty_record
+
+        return None
+
+    @classmethod
+    def create_late_submission_penalty(cls, obligation: ComplianceObligation) -> CompliancePenalty | None:
+        """
+        Calculate late submission penalty,
+        persist the penalty data to the database & generate an invoice in elicensing.
+
+        Args:
+            obligation: The compliance obligation object
+
+        Returns:
+            CompliancePenalty Record
+        """
+
+        # Refresh the elicensing data
+        ElicensingDataRefreshService.refresh_data_wrapper_by_compliance_report_version_id(
+            compliance_report_version_id=obligation.compliance_report_version_id
+        )
+
+        # Get the compliance deadline (Nov 30 of the year following the compliance period)
+        compliance_period = obligation.compliance_report_version.compliance_report.compliance_period
+        compliance_deadline = compliance_period.compliance_deadline
+
+        # Late submission penalty: Dec 1 (day after Nov 30)
+        late_submission_start_date = compliance_deadline + timedelta(days=1)
+
+        # Get supplementary submission date and calculate payment deadline (30 days after submission)
+        supplementary_submission_date = cls._get_last_supplementary_submission_date(obligation)
+        payment_deadline = supplementary_submission_date + timedelta(days=30)  # type: ignore[operator]
+
+        # Calculate late submission penalty
+        penalty_record = cls._calculate_late_submission_penalty(
+            obligation=obligation,
+            accrual_start_date=late_submission_start_date,
+            final_accrual_date=payment_deadline,  # type: ignore[arg-type]
```

Comment:
The final accrual date is not the payment_deadline, it is the date on which the obligation is paid to zero. Setting this to the payment_deadline of 30 days past the submission date ensures that this penalty stops accruing after 30 days whether it is paid or not.